conda_smithy/schema/models.py

```diff
@@ -0,0 +1,476 @@
+        ),
+    ] = False
+
+    docker: Annotated[
+        Union[CondaForgeDocker, None],
+        Field(
+            description="This is a mapping for Docker-specific configuration options."
+        ),
+    ] = CondaForgeDocker()
+
+    github: Annotated[
+        Union[GithubConfig, None],
+        Field(
+            description="Mapping for GitHub-specific configuration options",
+        ),
+    ] = GithubConfig(
+        user_or_org="conda-forge",
+        repo_name="",
+        branch_name="main",
+        tooling_branch_name="main",
+    )
+
+    idle_timeout_minutes: Annotated[
+        Union[int, None],
+        Field(
+            description="Configurable idle timeout.  Used for packages that don't have chatty enough builds. Applicable only to circleci and travis"
+        ),
+    ] = None
+
+    win: Annotated[
+        Union[WinConfig, None],
+        Field(
+            description="Windows-specific configuration options. This is largely an internal setting and should not normally be manually modified."
+        ),
+    ] = None
+
+    osx: Annotated[
+        Union[MacOsxConfig, None],
+        Field(
+            description="OSX-specific configuration options. This is largely an internal setting and should not normally be manually modified."
+        ),
+    ] = None
+
+    linux: Annotated[
+        Union[LinuxConfig, None],
+        Field(
+            description="Linux-specific configuration options. This is largely an internal setting and should not normally be manually modified."
+        ),
+    ] = None
+
+    linux_aarch64: Annotated[
+        Union[Aarch64Config, None],
+        Field(
+            description="ARM-specific configuration options. This is largely an internal setting and should not normally be manually modified."
+        ),
+    ] = None
+
+    linux_ppc64le: Annotated[
+        Union[Ppc64leConfig, None],
+        Field(
+            description="PPC-specific configuration options. This is largely an internal setting and should not normally be manually modified."
+        ),
+    ] = None
+
+    noarch_platforms: Annotated[
+        Union[List[Platforms], None],
+        Field(
+            description="Platforms on which to build noarch packages. The preferred default is a single build on linux_64."
+        ),
+    ] = [Platforms.linux_64.value]
+
+    os_version: Annotated[
+        Union[Dict[Platforms, str], None],
+        Field(
+            description="This key is used to set the OS versions for linux_* platforms. Valid entries map a linux platform and arch to either cos6 or cos7. Currently cos6 is the default for linux-64. All other linux architectures use CentOS 7."
+        ),
+    ] = {p.value: None for p in Platforms if p.value.startswith("linux")}
```

Comment:
We should use `default_factory` I think.

Reply:
thanks, I guess I didn't fully understand what it was